sched/irq/irq_attach.c

```diff
@@ -67,13 +68,13 @@ int irq_to_ndx(int irq)
 {
   DEBUGASSERT(g_irqmap_count < CONFIG_ARCH_NUSER_INTERRUPTS);
 
-  irqstate_t flags = spin_lock_irqsave(NULL);
+  irqstate_t flags = spin_lock_irqsave(&g_irq_attach_lock);
```

Comment:
atomic

Reply:
Due to the involvement of multiple global variables (g_irqmap[irq], g_irqmap_count) and their associated modification and conditional checks, I haven't found a suitable method to achieve atomic operations.